Integrations/integration-Cisco-umbrella.yml

```diff
@@ -0,0 +1,885 @@
+description: Cisco Umbrella Investigate
+detaileddescription: 'Cisco Investigate is part of the Cisco Umbrella package. When
+  you log into the CIsco Umbrella portal you will need to obtain the API Token for
+  the Cisco Investigate Feature. '
+configuration:
+- display: Please enter your api token
+  name: APIToken
+  defaultvalue: ""
+  type: 4
+  required: true
+- display: Use system proxy settings
+  name: proxy
+  defaultvalue: ""
+  type: 8
+  required: false
+- display: Do not validate server certificate (insecure)
+  name: insecure
+  defaultvalue: ""
+  type: 8
+  required: false
+script:
+  script: |2
+
+    var insecure = params.insecure;
+    var proxy = params.proxy;
+    var apitoken = params.APIToken;
+    var baseURL = 'https://investigate.api.umbrella.com';
+
+    var domainArg = (args.domain) ? args.domain : '';
+    domainArg = domainArg.replace(/^https?:\/\//i, '');
+
+    // removes 'http://'/'https://'' prefix
+    var domainWithoutProtocol = function(domain) {
+        return domain && domain.replace(/^https?:\/\//i, '');
+    }
+
+    // "one_two" to "One Two"
+    var keyToTableKey = function(s) {
+        return s && s.replace(/_/g, ' ')
+                .replace(/\w+/g, function(w){return w[0].toUpperCase() + w.slice(1).toLowerCase();});
+    }
+
+    // "one_two" to "OneTwo"
+    var keyToContextKey = function(s) {
+        return s && keyToTableKey(s).replace(/\s+/g, '');
+    }
+
+    var createTableEntry = function (name, contents, context, headers) {
+        return {
+            // type
+            Type: entryTypes.note,
+             // contents
+            ContentsFormat: formats.json, Contents: contents,
+            // human-readable
+            ReadableContentsFormat: formats.markdown, HumanReadable: tableToMarkdown(name, contents, headers),
+            // context
+            EntryContext: context
+        };
+    }
+
+    var startArg = (args.start) ? args.start : '';
```

Comment:
would batch all the xxxArg in the same area - startArg, domainArg. 
Would also document is it's not very clear in first sight what is their usage